File: T4D4-IU-cs50-problems-2022-x-project/98306330-main/cash/cash.c
#include <cs50.h>
#include <stdio.h>

int get_cents(void);
int calculate_quarters(int cents);
int calculate_dimes(int cents);
int calculate_nickels(int cents);
int calculate_pennies(int cents);

int main(void)
{
    // お客さまに何¢払えばいいのか聞く
    int cents = get_cents();

    // お客様にお渡しする25¢の枚数を計算する
    int quarters = calculate_quarters(cents);
    cents = cents - quarters * 25;

    // お客様にお渡しする10¢の枚数を計算する
    int dimes = calculate_dimes(cents);
    cents = cents - dimes * 10;

    // お客様に渡すべき5¢の枚数を計算する
    int nickels = calculate_nickels(cents);
    cents = cents - nickels * 5;

    //お客様にお渡しする1¢の枚数を計算する
    int pennies = calculate_pennies(cents);
    cents = cents - pennies * 1;

    // Sum coins
    int coins = quarters + dimes + nickels + pennies;

    // お客さまにお渡しするコインの総数を印刷する
    printf("%i\n", coins);
}

int get_cents(void)
{
    // 入力
    int cents;
    do
    {
        cents = get_int("Change owed:");
    }
    while (0 > cents);
    return cents;
}

int calculate_quarters(int cents)
{
    // 25¢が幾つ必要か
    int quarters;
    quarters = cents / 25;
    return quarters;

}

int calculate_dimes(int cents)
{
    // 10¢が幾つ必要か
    int dimes;
    dimes = cents / 10;
    return dimes;
}

int calculate_nickels(int cents)
{
    // 5¢が幾つ必要か
    int nickels;
    nickels = cents / 5;
    return nickels;
}

int calculate_pennies(int cents)
{
    // 1¢が幾つ必要か
    int pennies;
    pennies = cents / 1;
    return pennies;
}

File: T4D4-IU-cs50-problems-2022-x-project/98306330-main/sentimental-cash/cash.py
from cs50 import get_float

# decline
change = 0
coins = 0

while change <= 0:
    change = get_float("Change owed:")
    cents = round(change * 100)
    #  input

while cents > 0:
    if cents >= 25:
        cents = cents - 25
        coins = coins + 1
        # ここで1回先頭へ
    elif cents >= 10:
        cents = cents - 10
        coins = coins + 1
        # ここで1回先頭へ
    elif cents >= 5:
        cents = cents - 5
        coins = coins + 1
        # ここで1回先頭へ
    elif cents >= 1:
        cents = cents - 1
        coins = coins + 1
        # ここで1回先頭へ

print(coins)